components/slideshow.tsx

```diff
@@ -0,0 +1,94 @@
+'use client';
+import Image from 'next/image';
+import { useCallback, useEffect, useState } from 'react';
+import { FaChevronLeft, FaChevronRight } from 'react-icons/fa';
+
+export default function Slideshow({ images }: { images: string[] }) {
+  const [currentSlide, setCurrentSlide] = useState<number>(0);
+  const [hovered, setHovered] = useState<boolean>(false);
+
+  const prevSlide = useCallback(() => {
+    setCurrentSlide((prev: number) =>
+      prev === 0 ? images.length - 1 : prev - 1
+    );
+  }, [images.length]);
+
+  const nextSlide = useCallback(() => {
+    setCurrentSlide((prev: number) =>
+      prev === images.length - 1 ? 0 : prev + 1
+    );
+  }, [images.length]);
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      nextSlide();
+    }, 3000);
+
+    return () => clearInterval(intervalId);
+  }, [nextSlide]);
+
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (hovered) {
```

Comment:
No, do `if (!hovered) return;`. Doing it this way unnecessarily increases indentation.